support/metrics/metricsTracker.go

```diff
@@ -190,34 +188,65 @@ func MakeMetricsTracker(
 	}, nil
 }
 
+// MakeMetricsTrackerGui is a factory method to create a `metrics.Tracker` from the CLI.
+func MakeMetricsTrackerGui(
+	userID string,
+	deviceID string,
+	apiKey string,
+	client *http.Client,
+	botStartTime time.Time,
+	version string,
+	goos string,
+	goarch string,
+	goarm string,
+	guiVersion string,
+) (*MetricsTracker, error) {
+	props := commonProps{
+		CliVersion: version,
+		Goos:       goos,
+		Goarch:     goarch,
+		Goarm:      goarm,
+		GuiVersion: guiVersion,
+	}
+
+	return &MetricsTracker{
+		client:       client,
+		apiKey:       apiKey,
+		userID:       userID,
+		deviceID:     deviceID,
+		props:        props,
+		botStartTime: botStartTime,
+	}, nil
+}
+
 // GetUpdateEventSentTime gets the last sent time of the update event.
 func (mt *MetricsTracker) GetUpdateEventSentTime() *time.Time {
 	return mt.updateEventSentTime
 }
 
 // SendStartupEvent sends the startup Amplitude event.
 func (mt *MetricsTracker) SendStartupEvent() error {
-	return mt.sendEvent(startupEventName, mt.props)
+	return mt.SendEvent(startupEventName, mt.props)
 }
 
 // SendUpdateEvent sends the update Amplitude event.
 func (mt *MetricsTracker) SendUpdateEvent(now time.Time, success bool, millisForUpdate int64) error {
-	commonProps := mt.props
-	commonProps.SecondsSinceStart = now.Sub(mt.botStartTime).Seconds()
+	mt.props.SecondsSinceStart = now.Sub(mt.botStartTime).Seconds()
```

Comment:
why are we setting this on the `mt.props` vs on a new copy of the props object? I feel like the previous way was better with `commonProps`.

(will need to check the golang semantics of where/when we create a copy of structs)

Reply:
discussed on call, decided to move mt.props.SecondSinceStart into `SendEvent` because it's common to all events. Callers should pass in a `now` field.